src/riak_kv_qry_buffers.erl

```diff
@@ -420,55 +427,97 @@ do_batch_put(QBufRef, Data, #state{qbufs          = QBufs0,
             end
     end.
 
--spec maybe_add_chunk(#qbuf{}, [data_row()],
-                      non_neg_integer(), non_neg_integer()) ->
-                             {ok, #qbuf{}, non_neg_integer()} | {error, total_qbuf_size_limit_reached | riak_kv_qry_buffers_ldb:errors()}.
-maybe_add_chunk(#qbuf{ldb_ref       = LdbRef,
-                      orig_qry      = ?SQL_SELECT{'ORDER BY' = OrderBy},
-                      chunks_got    = ChunksGot0,
-                      chunks_need   = ChunksNeed,
-                      size          = Size,
-                      total_records = TotalRecords0,
-                      key_field_positions = KeyFieldPositions} = QBuf0,
-                Data,
-                TotalSize, HardWatermark) ->
+-spec add_chunk(#qbuf{}, [data_row()],
+                non_neg_integer(), non_neg_integer(), non_neg_integer()) ->
+                       {ok, #qbuf{}, non_neg_integer()} | {error, total_qbuf_size_limit_reached | riak_kv_qry_buffers_ldb:errors()}.
+add_chunk(#qbuf{ldb_ref       = LdbRefOrDelayedCreateFun,
+                inmem_buffer  = InmemBuffer0,
+                orig_qry      = ?SQL_SELECT{'ORDER BY' = OrderBy},
+                chunks_got    = ChunksGot0,
+                chunks_need   = ChunksNeed,
+                size          = Size,
+                total_records = TotalRecords0,
+                key_field_positions = KeyFieldPositions} = QBuf0,
+          Data,
+          TotalSize, HardWatermark, InmemMax) ->
     ChunkSize = compute_chunk_size(Data),
     case TotalSize + ChunkSize > HardWatermark of
         true ->
             {error, total_qbuf_size_limit_reached};
         false ->
+            %% update stats/counters
+            ChunkId = ChunksGot0,
+            ChunksGot = ChunksGot0 + 1,
```

Comment:
The comment after this line is a few lines down from where it should be. Also the tone is odd: does it mean that currently the ID is *not* the serial number of the subquery? Please make it more definitive.